### ISSUE TITLE
Add user mapper that supports users that already exists

### ISSUE THREAD

**Issue by timonegk** [fsinfuhh/py_simple_openid_connect#17]
This is useful when adding the library to an existing project and existing users have to be mapped. The default user mapper always creates new user objects, potentially violating unique constraints in the database. I have added a user mapper that supports this to two applications (karma and bitpoll):

https://github.com/fsinfuhh/Bitpoll/blob/1e51ef89dcf0647d9b50285ba643ba71641a2cde/bitpoll/base/openid.py#L14-L29

I think this should be integrated with the library.

**Comment by ftsell** [comment]
Yes this might be a good idea. I'm not sure if we want to enable it by default since this has potential security implications that might surprise someone using the library? Maybe we could ship two UserManager implementations: One that creates new user objects and one that updates existing ones. Then a programmer would have to decide which one works better for their usecase.
